lte/gateway/c/session_manager/LocalEnforcer.cpp

```diff
@@ -1944,29 +1953,18 @@ void LocalEnforcer::schedule_revalidation(
 }  // namespace magma
 
 void LocalEnforcer::handle_activate_ue_flows_callback(
-    const std::string& imsi, const std::string& ip_addr,
-    const std::string& ipv6_addr, const Teids teids, Status status,
-    ActivateFlowsResult resp) {
+    const std::string& imsi, const std::string& session_id,
+    const std::string& ip_addr, const std::string& ipv6_addr, const Teids teids,
+    Status status, ActivateFlowsResult resp) {
   if (status.ok()) {
-    MLOG(MDEBUG) << "Pipelined add ue enf flow succeeded for " << imsi;
+    MLOG(MDEBUG) << "Pipelined add ue enf flow succeeded for " << session_id;
     return;
   }
 
-  MLOG(MERROR) << "Could not activate rules for " << imsi
+  MLOG(MERROR) << "Could not activate rules for " << session_id
                << ", rpc failed: " << status.error_message()
                << ", terminating session...";
 
-  SessionSearchCriteria criteria(imsi, IMSI_AND_UE_IPV4_OR_IPV6, ip_addr);
-  auto session_map = session_store_.read_sessions({imsi});
```

Comment:
race condition may happen here

Reply:
ah is this outside of the eventloop?

Reply:
I think so becasuse this is called by pieplined_client. Which i think it runs on a different thread (at least sessiond was showing different thread id on logs)

here is where it calls 

https://sourcegraph.com/-/editor?remote_url=git%40github.com%3Amagma%2Fmagma.git&branch=master&file=lte%2Fgateway%2Fc%2Fsession_manager%2FLocalEnforcer.cpp&editor=JetBrains&version=v1.2.1&utm_product_name=CLion&utm_product_version=2021.1.2&start_row=367&start_col=10&end_row=367&end_col=10